### PR TITLE
Add documentation for entity and device filters

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add documentation for new `filter` option for `entity` and `device` selectors. Support for multiple device_classes has been added too.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/87536 https://github.com/home-assistant/frontend/pull/15302
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/blueprint/selectors.markdown [hunk 1]
```diff
@@ -397,8 +399,35 @@ entity:
...
+        with a single device_class, or a list of string device_class to limit
+        the selection to.
+      type: [device_class, list]
+      required: false
+filter:
+  description: >
+    When filter options are provided, the list of devices is filtered by devices
+    that at least provide one entity that matches the given conditions. Can be either
+    a object or a list of object.
+  type: map
```
Comment (piitaya) [inline source/_docs/blueprint/selectors.markdown]: Seems that `[map, list]` is not allowed so I set `map`.

File: source/_docs/blueprint/selectors.markdown [hunk 2]
```diff
@@ -433,9 +448,10 @@ And this is what is looks like in YAML:
 
 ```yaml
 device:
-  integration: deconz
-  manufacturer: Philips
-  model: RWL021
+  filter:
+    integration: deconz
+    manufacturer: Philips
+    model: RWL021
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown]:
```suggestion
    - integration: deconz
      manufacturer: Philips
      model: RWL021
```

File: source/_docs/blueprint/selectors.markdown [hunk 3]
```diff
@@ -556,9 +581,10 @@ And this is what it looks like in YAML:
 ```yaml
 entity:
   multiple: true
-  integration: zha
-  domain: binary_sensor
-  device_class: motion
+  filter:
+    integration: zha
+    domain: binary_sensor
+    device_class: motion
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown]:
```suggestion
    - integration: zha
      domain: binary_sensor
      device_class: motion
```

**Review by frenck — CHANGES_REQUESTED** [review]
Couple of small corrections to match our style guides:

<https://developers.home-assistant.io/docs/documenting/yaml-style-guide#properties-that-accept-a-mapping-or-a-list-of-mappings>

**Review by frenck — APPROVED** [review]
Thanks, @piitaya 👍 

../Frenck

**Pull request merged**